src/protos/metadata.proto

```diff
@@ -4,6 +4,16 @@ package protos;
 
 import "src/protos/chunk_server.proto";
 
+// Metadata of a given file. The reason that we define the file metadata in proto is
+// that later we will use the protobuf to serialize / deserialize this data to
+// checkpoint / log files
```

Comment:
I think it should be per update for logs because that is what it meant to do. Defining this state in proto at least would help when we load the checkpoint. We can use the serialize function in the generated code for each field. I'd have to think more about how to realize this but that is in phase 2.